timemachine/md/builders.py

```diff
@@ -161,17 +163,39 @@ def build_protein_system(
     assert modeller.getTopology().getNumAtoms() == solvated_host_coords.shape[0]
 
     print("building a protein system with", num_host_atoms, "protein atoms and", num_water_atoms, "water atoms")
-    solvated_host_system = host_ff.createSystem(
+    solvated_omm_host_system = host_ff.createSystem(
         modeller.topology, nonbondedMethod=app.NoCutoff, constraints=None, rigidWater=False
     )
 
+    # note, masses are not needed....
+    (bond, angle, proper, improper, nonbonded), masses = openmm_deserializer.deserialize_system(
+        solvated_omm_host_system, cutoff=1.2
+    )
+
+    solvated_host_system = HostSystem(
+        bond=bond,
+        angle=angle,
+        proper=proper,
+        improper=improper,
+        nonbonded_all_pairs=nonbonded,
+    )
+
     # Determine box from the system's coordinates
     box = get_box_from_coords(solvated_host_coords)
 
-    return solvated_host_system, solvated_host_coords, box, modeller.topology, num_water_atoms
+    assert len(list(modeller.topology.atoms())) == len(solvated_host_coords)
```

Comment:
@mcwitt added this assertion for sanity checking purposes going forward (we should probably make sure that the `CIFWriter` also does a similar check...